feature/gnoi/system/tests/chassis_reboot_status_and_cancel_test/chassis_reboot_status_and_cancel_test.go

```diff
@@ -0,0 +1,177 @@
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	spb "github.com/openconfig/gnoi/system"
+	"github.com/openconfig/ondatra"
+)
+
+const (
+	oneMinuteInNanoSecond = 6e10
+	rebootDelay           = 120
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Test cases:
+//  1) Send gNOI reboot status request.
+//   - Check the reboot status before sending reboot request.
+//     - Verify the reboot status is not active.
+//   - Check the reboot status after sending reboot request.
+//     - Verify the reboot status is active.
+//     - Verify the reason from reboot status response matches reboot message.
+//     - Verify the wait time from reboot status response matches reboot delay.
+//  2) Cancel gNOI reboot request.
+//   - Cancel reboot request before the test
+//     - Verify that there is no response error returned.
+//   - Send reboot request with delay.
+//     - Verify the reboot status is active.
+//   - Send reboot cancel request.
+//     - Verify the reboot status is not active.
+//
+// Topology:
+//   dut:port1 <--> ate:port1
+//
+// Test notes:
+//  - gnoi operation commands can be sent and tested using CLI command grpcurl.
+//    https://github.com/fullstorydev/grpcurl
+//
+
+func TestRebootStatus(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+
+	cases := []struct {
+		desc          string
+		rebootRequest *spb.RebootRequest
+		rebootActive  bool
+		cancelReboot  bool
+	}{
+		{
+			desc:          "no reboot requested",
+			rebootRequest: nil,
+			rebootActive:  false,
+		},
+		{
+			desc: "reboot requested with delay",
+			rebootRequest: &spb.RebootRequest{
+				Method:  spb.RebootMethod_COLD,
+				Delay:   rebootDelay * oneMinuteInNanoSecond,
+				Message: "Reboot chassis with delay",
+				Force:   true,
+			},
+			rebootActive: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			if tc.rebootRequest != nil {
+				t.Logf("Send reboot request: %v", tc.rebootRequest)
+				rebootResponse, err := gnoiClient.System().Reboot(context.Background(), tc.rebootRequest)
+				t.Logf("Got reboot response: %v, err: %v", rebootResponse, err)
+				if err != nil {
+					t.Fatalf("Failed to request reboot with unexpected err: %v", err)
+				}
+			}
+
+			resp, err := gnoiClient.System().RebootStatus(context.Background(), &spb.RebootStatusRequest{})
+			t.Logf("DUT rebootStatus: %v, err: %v", resp, err)
+			if err != nil {
+				t.Fatalf("Failed to get reboot status with unexpected err: %v", err)
+			}
+			if resp.GetActive() != tc.rebootActive {
+				t.Errorf("resp.GetActive(): got %v, want %v", resp.GetActive(), tc.rebootActive)
+			}
+
+			if tc.rebootRequest != nil {
+				if resp.GetReason() != tc.rebootRequest.GetMessage() {
+					t.Errorf("resp.GetReason(): got %v, want %v", resp.GetReason(), tc.rebootRequest.GetMessage())
+				}
+				if resp.GetWait() > tc.rebootRequest.GetDelay() {
```

Comment:
Shouldn't this be resp.GetWait() <= tc.RebootRequest.GetDelay()? I don't think the time left should ever be more than the delay requested ([ref](https://github.com/openconfig/gnoi/blob/6bddacd72ef8aa06304b887df414a43c8e63ff27/system/system.proto#L153)).

Reply:
Right. The test would fail with err msg if resp.GetWait() > tc.rebootRequest.GetDelay()

				if resp.GetWait() > tc.rebootRequest.GetDelay() {
					t.Errorf("resp.GetWait(): got %v, want <= %v", resp.GetWait(), tc.rebootRequest.GetDelay())
				}

Reply:
Whoops I was confusing myself.  Right.